### DISCUSSION TITLE
Incorrect geolocation in thousands of photos

### DISCUSSION THREAD

**Discussion by alfondotnet** [immich-app/immich#12224] [General]
Hi fam!
Have an issue where thousands of my photos have the GPS longitude inverted, for example:

this photo shows through exiftool:

```
GPS Longitude                   : 122 deg 27' 53.79" W
GPS Longitude Ref               : West
GPS Latitude                    : 37 deg 40' 35.39" N
GPS Latitude Ref                : North
```

And that's being read as:

```
37.6764966666667,122.464941666667
```

in the DB's `exif` table.

It should be:

```
37.6764966666667,-122.464941666667
```

So it shows in China instead of California.

Now, If I download the photo and reupload through the WebUI, the metadata will get fixed but the "date" will be reset to the upload date plus is tedious to do with thousands of photos.

Any tips?

Thanks!

**Comment by alextran1502** [comment]
Can you attach one of the file with issue? Which device are these photos coming from?

**Reply by alfondotnet** [reply]
Here it is:

https://drive.google.com/file/d/1-Y8zeefmk6rsFxLpEsrnLsSpQQVmQoIW/view?usp=sharing

it was made with an iPhone 13

**Comment by zackpollard** [comment]
Hey, if these are fixed when uploaded through the web UI, maybe it was an old bug that is now fixed. Could you try opening one of the problematic photos, then clicking the context menu in the top right on web and click "refresh metadata"? If this fixes it then you could just re-run extraction for your entire library.

**Comment by alfondotnet** [comment]
I tried that and it doesn’t work, which is odd, unfortunately

**Reply by alfondotnet** [reply]
@zackpollard , what's interesting is that, if I delete the `exif` entry and "refresh the metadata", it will regenerate wrong again, yet if I delete the asset and reupload it will be fine, then. Anything that comes to mind as to what would be different, perhaps some other data cached somewhere?

Ideally I'd love to do a massive delete of exif data and regenerate metadata, seems like the easiest path

**Reply by zackpollard** [reply]
Please check if you have xmp files next to the file with incorrect data, if you do, check if the contents of that file includes a different location.

**Reply by alfondotnet** [reply]
Awesome!! 

The XMP sidecar files weren't colocated for some reason but looking at the DB, the busted photos had sidecar files in a totally different `/upload` subdirectory (alongside different assets) and those contained the busted locations. So I just did a big pass of wiping the `sidecarPath` column of these assets and re-ran the global metadata extraction job. 

Stoked!! Thank you!

**Reply by zackpollard** [reply]
No worries, glad you got it sorted! 😉
